go/parquet/file/file_writer.go

```diff
@@ -165,7 +166,9 @@ func (fw *Writer) Close() error {
 			fw.rowGroupWriter.Close()
 		}
 		fw.rowGroupWriter = nil
-		defer fw.sink.Close()
+		defer func() {
+			err = errors.Join(err, fw.sink.Close())
```

Comment:
unless.. I can't use `errors.Join` (go 1.20), since I see the build fails on a few machines, can I upgrade the go version in the go git workflow as well?

Reply:
I think the project policy is 2 versions behind for compatibility

Reply:
fair enough, I have rewritten the logic as requested.